sig-network/link-state-proposal.md

```diff
@@ -0,0 +1,197 @@
+
+virt-launcher's `LibvirtDomainManager.SyncVMI` will be extended to support updating the interfaces link state while the VM is running.
+
+virt-launcher's NIC hotplug logic will be adjusted to support hot-plugging a NIC while its link state is down.
+
+### Network Binding Plugins Support
+
+When using the functionality purposed by this enhancement with network binding plugins using a sidecar container -
+its up to the plugin's vendor to support this functionality - as they control the interface section in the domain XML.
+
+### Link State Reporting
+
+A new field will be added to the VMI interface status that will be used for reporting the interface's current link state (up/down):
+```go
+type VirtualMachineInstanceNetworkInterface struct {
+	...
+	LinkState string `json:"linkState,omitempty"`
+	...
+}
+```
+
+virt-handler will be responsible for reporting this field's value, based on domain information received from virt-launcher.
+
+## API Examples
+### Controlling The Link State
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: my-vm
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+            - name: default
+              state: down
+              masquerade: { }
+      networks:
+        - name: default
+          pod: { }
+```
+### Link State Reporting
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: my-vm
+spec:
+  domain:
+    devices:
+      interfaces:
+      - name: default
+        state: down
+        masquerade: { }
+  networks:
+  - name: default
+    pod: { }
+status:
+  interfaces:
+    - name: default
+      linkState: down
+```
+
+## Alternatives
+
+## Scalability
+
+This enhancement does not affect scalability.
+
+## Update/Rollback Compatibility
+
+As this functionality will require the cooperation of the virt-launcher component, it could only be supported by
+up-to-date virt-launcher pods.
+
+## Functional Testing Approach
+
+The following e2e scenarios will be added:
+- A VM started with interface with a link state set to `down`, then changed to `up`.
+- Setting a link state of a running VM to `down`, then `up`.
+- Hot-plugging an interface connected to a secondary network with link state set to `down`, then changed to `up`.
+- Migrating a VM that has an interface with a link state set to `down`, then changed to `up`.
+
+## Implementation Phases
+
+1. Addition of the new `InterfaceState` values.
+2. Adjusting the network validator to accept the new `InterfaceState` values.
+3. Adjusting virt-launcher:
+   - On creation
+   - When VM is running
+   - On hotplug
+4. Addition of e2e tests.
+5. Addition of the `LinkState` field to `VirtualMachineInstanceNetworkInterface`.
+6. Adjusting virt-handler's reporting logic.
+7. Adjusting e2e tests.
+
+## Feature lifecycle Phases
+
+### GA
+Since this feature is small and has no significant risk, it could be released directly as GA.
```

Comment:
Let's test it a little before determining that... :) 
An innocent-looking feature might turn out to be a hidden monster...
And specifically with this feature - the ability to set the guest's pod networking down worries me.

Reply:
> Let's test it a little before determining that... :)

E2E tests are planned, and the test case / scenarios are written down.
Do you think additional scenarios should be added?

Reply:
> And specifically with this feature - the ability to set the guest's pod networking down worries me.

Could you please elaborate why does it worries you?
The user has full control over this field, and the system has none - so it shouldn't automatically change.